File: app/src/main/java/com/raival/fileexplorer/tab/file/observer/FileListObserver.java
package com.raival.fileexplorer.tab.file.observer;

import androidx.recyclerview.widget.RecyclerView;

import com.raival.fileexplorer.tab.file.FileExplorerTabFragment;
import com.raival.fileexplorer.tab.file.adapter.FileListAdapter;

public class FileListObserver extends RecyclerView.AdapterDataObserver {
    private final FileExplorerTabFragment parentFragment;
    private final FileListAdapter fileListAdapter;

    public FileListObserver(FileExplorerTabFragment fragment, FileListAdapter adapter) {
        parentFragment = fragment;
        fileListAdapter = adapter;
        checkIfEmpty();
    }

    @Override
    public void onChanged() {
        super.onChanged();
        checkIfEmpty();
    }

    private void checkIfEmpty() {
        parentFragment.showPlaceholder(fileListAdapter != null && fileListAdapter.getItemCount() == 0);
    }
}

File: app/src/main/java/com/raival/fileexplorer/tab/file/observer/FileListObserver.kt
package com.raival.fileexplorer.tab.file.observer

import androidx.recyclerview.widget.RecyclerView.AdapterDataObserver
import com.raival.fileexplorer.tab.file.FileExplorerTabFragment
import com.raival.fileexplorer.tab.file.adapter.FileListAdapter

class FileListObserver(
    private val parentFragment: FileExplorerTabFragment,
    private val fileListAdapter: FileListAdapter?
) : AdapterDataObserver() {

    override fun onChanged() {
        super.onChanged()
        checkIfEmpty()
    }

    private fun checkIfEmpty() {
        parentFragment.showPlaceholder(fileListAdapter != null && fileListAdapter.itemCount == 0)
    }

    init {
        checkIfEmpty()
    }
}